common/src/main/java/io/netty/util/internal/RecyclableArrayList.java

```diff
@@ -40,6 +40,8 @@ protected RecyclableArrayList newObject(Handle handle) {
         }
     };
 
+    private boolean added;
```

Comment:
not sure as we only update this on add and not on remove. Maybe insertSinceRecycled ?